test/models/artefact_action_test.rb

```diff
@@ -39,86 +39,96 @@ def base_fields
     }
   end
 
+  def setup
+    @artefact = Artefact.create!(base_fields)
```

Comment:
It's better to do the refactoring of the tests in a separate commit, so that the commit is easier to review. Currently it's hard to see which changes are relevant for this PR.